v1/ast/term.go

```diff
@@ -2540,6 +2531,9 @@ func (obj *object) get(k *Term) *objectElem {
 	case Number:
 		if xi, ok := x.Int64(); ok {
 			equal = func(y Value) bool {
+				if x == y {
+					return true
+				}
```

Comment:
We don't do this in the `insert()` `equal` func. That's different somehow? (and is that why these aren't reused?)